### PR TITLE
SCS-0001: Sovereign Cloud Standard

### PR DESCRIPTION
**Pull request by horazont**
This document provides a process for managing the standards and similar documents developed within the SCS community.

### CODE REVIEW

File: Standards/scs-0001-sovereign-cloud-standards.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+The lifecycle of an SCS document goes through the following phases:
+Proposal, Draft, Stable, Deprecated, and Rejected.
+
+All decisions for phase transitions follow loose consensus,
+where the group which has to form the consensus depends on the `track` of the document:
+
+- IaaS: The team working on infrastructure-as-a-service topics (formerly Team 1)
+- KaaS: The team working on Kubernetes-as-a-service topics (formerly Team 2)
+- Ops: The team working on operations topics (formerly Team 3)
```
Comment (fkr) [inline Standards/scs-0001-sovereign-cloud-standards.md:128]: Please let's drop the 'formerly Team X'.

Comment (horazont) [reply Standards/scs-0001-sovereign-cloud-standards.md:128]: Ah good suggestion, I was already wondering what to do about IAM.

File: Standards/scs-0001-sovereign-cloud-standards.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+This major version number is a positive number
+and the numbering starts at one for each document.
+
+In order to make organisation of the SCS documents easier,
+each document also has a slugified title.
+The slugified title MUST NOT be changed after the acceptance of the document into the repository,
+as it is part of its canonical URL.
+It MUST consist only of lower-case ASCII letters, numbers and hyphens.
+It MUST NOT start with a hyphen and SHOULD start with a lower-case letter.
+It SHOULD NOT contain more than one subsequent hyphen and SHOULD start with a lower-case letter.
```
Comment (joshmue) [inline Standards/scs-0001-sovereign-cloud-standards.md:76]: `SHOULD start with a lower-case letter` is a duplicate of the second part of the prior sentence here, isn't it?

File: Standards/scs-0001-sovereign-cloud-standards.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+the resulting file name would be `scs-0390-v2-flavor-naming.md`.
+
+In addition to the number, each document has the following metadata,
+embedded in the markdown header.
+
+| Field name | Requirement | Description |
+| --- | --- | --- |
+| `type` | REQUIRED | one of `Procedural`, `Standard`, or `Decision Record` |
+| `status` | REQUIRED | one of `Draft`, `Stable`, `Deprecated`, or `Rejected` |
+| `track` | REQUIRED | one of `Global`, `IaaS`, `Ops`, `KaaS` |
```
Comment (joshmue) [inline Standards/scs-0001-sovereign-cloud-standards.md:93]: `IAM` should be appended, like below.

File: Standards/scs-0001-sovereign-cloud-standards.md [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+In the proposal phase,
+the document number MUST be replaced with `xxxx` in the file name.
+The major version MUST be 1.
+
+For a document with a slugified title `flavor-naming`,
+the file name would for instance be `scs-xxxx-v1-flavor-naming.md`.
+
+The metadata MUST indicate the intended `track` and `type` of the document,
+and the `status` MUST be set to `Draft`
+and the major number in the `version` SHOULD be `0`.
```
Comment (joshmue) [inline Standards/scs-0001-sovereign-cloud-standards.md:151]: I'm a bit unsure whether there is a difference between the major version in the file name which MUST be 1 on the one hand, and the major number in `version` which SHOULD be `0`. Are they major versions/numbers with distinct meaning?

Furthermore, is `version` a metadata field in the markdown headers? If yes, it is missing in the above table of metadata fields.

Comment (horazont) [reply Standards/scs-0001-sovereign-cloud-standards.md:151]: Good catch. This is a leftover, there should be no version field at all anymore (I had a previous iteration with a separate version field and that's gone now, I just missed that bit of text).

File: Standards/scs-0001-sovereign-cloud-standards.md (hunk 3 above)
Comment (horazont) [reply Standards/scs-0001-sovereign-cloud-standards.md]: Fixed thanks.

File: Standards/scs-0001-sovereign-cloud-standards.md (hunk 2 above)
Comment (horazont) [reply Standards/scs-0001-sovereign-cloud-standards.md]: Fixed, thanks!

File: Standards/scs-0001-sovereign-cloud-standards.md [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+Proposal, Draft, Stable, Deprecated, and Rejected.
+
+All decisions for phase transitions follow loose consensus,
+where the group which has to form the consensus depends on the `track` of the document:
+
+- IaaS: The team working on infrastructure-as-a-service topics
+- KaaS: The team working on Kubernetes-as-a-service topics
+- Ops: The team working on operations topics
+- IAM: The team working on idenity and access management topics
+- Global: The entire SCS community.
```
Comment (fkr) [inline Standards/scs-0001-sovereign-cloud-standards.md:130]: no trailing dot please.

**Review by joshmue — DISMISSED** [review]

**Review by fkr — CHANGES_REQUESTED** [review]
@horazont please see my last comment, about this one trailing dot ;)

File: Standards/scs-0001-sovereign-cloud-standards.md (hunk 5 above)
Comment (horazont) [reply Standards/scs-0001-sovereign-cloud-standards.md]: Fixed.

File: Standards/scs-0001-sovereign-cloud-standards.md [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+Sometimes during the development of the SCS standard,
+a complex technical decision needs to be taken,
+which does not directly result in a new standard.
+
+To keep a record of such decisions for future reference,
+they were formerly documented in a format called
+"Architecture Decision Records" (ADRs).
+
+The SCS document format supersedes and integrates such ADRs
+as documents of type `Decision Record`.
```
Comment (fkr) [inline Standards/scs-0001-sovereign-cloud-standards.md:55]: I suggest to rephrase this, as we never really used ADRs, so imho there is no need to refer to them as.
Suggestion:

Sometimes during the development of the SCS standard, a complex technical decision needs to be taken, which does not directly result in a new standard.
The SCS document format integrates the documentation of such decisions with the type `Decision Record`.

File: Standards/scs-0001-sovereign-cloud-standards.md [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+# Requirements
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+In addition, "FORBIDDEN" is to be interpreted equivalent to "MUST NOT".
+
+# Motivation and Objectives
+
+TBD.
+
```
Comment (fkr) [inline Standards/scs-0001-sovereign-cloud-standards.md:27]: imho this can be safely removed.

**Review by fkr — APPROVED** [review]
Added couple more comments. These can be adjusted in the PR or after it being merged (as such, I'm still 👍 the PR)

File: Standards/scs-0001-sovereign-cloud-standards.md (hunk 7 above)
Comment (horazont) [reply Standards/scs-0001-sovereign-cloud-standards.md]: Addressed.

File: Standards/scs-0001-sovereign-cloud-standards.md (hunk 6 above)
Comment (horazont) [reply Standards/scs-0001-sovereign-cloud-standards.md]: Addressed.

**Review by fkr — APPROVED** [review]
LGTM! Thanks @horazont for all the work!

**Pull request merged**